packages/nimble-components/src/table/specs/table-column-specs/table-column-text-field.md

```diff
@@ -0,0 +1,175 @@
+
+The visual appearance of the text content will match that of a `nimble-text-field`.
+
+---
+
+## Implementation
+
+For the `cellTemplate` implementation required for a `TableColumn<>` implementation we will provide something similar to the following:
+
+```TS
+public class TableColumnTextField ...
+{
+    ...
+
+    public cellDataKeyNames = ['value'] as const;
+
+    public readonly cellStyles: ElementStyles => css`
+        .text-value {
+            // set necessary text-value styles
+        }
+
+        .placeholder {
+            // set necessary placeholder styles
+        }
+    `;
+
+    public readonly cellTemplate: ViewTemplate<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>> =>
+        html<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>>`
+            ${when(x => x.data.value, html`
+                <span class="text-value">{x => x.data.value}</span>
+            `)}
+            ${when(x => !x.data.value, html`
+                <span class="placeholder">{x => x.columnConfig.placeholder}</span>
+            `)}
+        `;
+
+    ...
+}
+```
+
+### Alternatives considered
+
+We are using `span` elements for the text rendering of the values instead of a `nimble-text-field` for performance's sake, as we avoid the notably heavier cost of using a custom element. The other benefits that using a `nimble-text-field` offer, such as built-in styling and a placeholder implementation seem trivial to replicate, which seems worth it for a performance improvement.
+
+One notable difference in behavior is that the proposed implementation will not support a behavior present in the `nimble-text-field` where a user can begin dragging at an arbitrary location in the text, and go to the end of the text, even if it has been clipped by the column (and thus showing an `...`). Users will still be able to double-click such text and copy the entire contents.
+
+### States
+
+N/A
+
+### Accessibility
+
+Accessibility of the cells rendered using the `nimble-table-column-text-field` are handled via the [`nimble-table-cell`](https://github.com/ni/nimble/blob/f663c38741e731bef91aa58e8fb2d1cec653b679/packages/nimble-components/src/table/components/cell/template.ts#L6) which has a `role` of [`cell`](https://w3c.github.io/aria/#cell).
+
+### Globalization
+
+None
+
+### Security
+
+None
+
+### Performance
+
+As discussed in the implementation section we are making a deliberate choice to not use the `nimble-text-field` to provide the best initialization performance possible.
+
+### Dependencies
+
+None.
+
+### Test Plan
+
+Unit tests will be written to test the component.
+
+### Tooling
+
+None.
+
+### Documentation
+
+This component will be documented via its usage in the storybook for the `nimble-table`.
+
+---
+
+## Open Issues
+
+_Highlight any open questions for discussion during the spec PR. Before the spec is approved these should typically be resolved with the answers being incorporated in the spec document._
```

Comment:
Should there be an open question about exposing styling options? For example, should we provide configuration for making placeholders italics? I think this is a question for designers, but depending on their answer, we might need to modify this API.

Reply:
Added an open question. @nate-ni, it would be good to get your feedback on that open question.